gh-106318: Add an example for `str.capitalize()`

blaisep opened the pull request:
This is the smallest possible step toward  #106318 , adding examples for the string methods. The content was  contributed by @adorilson in #105670 Because  https://github.com/python/cpython/pull/119445 is big and difficult to review, this PR will be updated as the individual examples get merged.

<!--
Thanks for your contribution!
Please read this comment in its entirety. It's quite important.

# Pull Request title

It should be in the following format:

```
gh-NNNNNN: Summary of the changes made
```

Where: gh-NNNNNN refers to the GitHub issue number.

Most PRs will require an issue number. Trivial changes, like fixing a typo, do not need an issue.

# Backport Pull Request title

If this is a backport PR (PR made against branches other than `main`),
please ensure that the PR title is in the following format:

```
[X.Y] <title from the original PR> (GH-NNNNNN)
```

Where: [X.Y] is the branch name, for example: [3.13].

GH-NNNNNN refers to the PR number from `main`.

-->


<!-- readthedocs-preview cpython-previews start -->
----
📚 Documentation preview 📚: https://cpython-previews--134272.org.readthedocs.build/en/134272/library/stdtypes.html#str.capitalize

<!-- readthedocs-preview cpython-previews end -->

<!-- gh-issue-number: gh-106318 -->
* Issue: gh-106318
<!-- /gh-issue-number -->

blaisep committed:
Add an example for `str.capitalize` for #106318

This is the smallest possible step toward adding examples for the string methods.
The content was  contributed by @adorilson in #105670
Because  https://github.com/python/cpython/pull/119445 is big and difficult to review,
this PR will be updated as the individual examples get merged.

StanFromIreland commented on Doc/library/stdtypes.rst:
```diff
@@ -1759,7 +1759,14 @@ expression support in the :mod:`re` module).
+      >>> 'PYTHON IS AMAZING'.capitalize()
+      'Python is amazing'
+      >>> 'ǋemačka Starts With a non-english Digraph'.capitalize()
+      'ǋemačka starts with a non-english digraph'
```
Maybe an example with a bigger visual difference?

Maybe
```
'Ǉjubljana starts With a non-english digraph'.capitalize()
```
since it is then `ǈ`

blaisep replied:
@StanFromIreland , oooh I see what you mean, thank you.

blaisep committed:
Replace the city name in the example.

The suggestion was to use a example where there was a bigger contrast between the uppercase in the input stream and the mixed case in the result.

blaisep committed:
Merge branch 'main' into 106318

hugovk commented on Doc/library/stdtypes.rst:
```diff
@@ -1759,7 +1759,14 @@ expression support in the :mod:`re` module).
+      >>> 'PYTHON IS AMAZING'.capitalize()
+      'Python is amazing'
+      >>> 'Ǉjubljana starts With a non-english digraph'.capitalize()
+      'ǈjubljana starts with a non-english digraph'
```
```suggestion
      >>> 'Ǉjubljana starts With a non-English digraph'.capitalize()
      'ǈjubljana starts with a non-english digraph'
```

hugovk replied:
It doesn't look like the word Ljubljana is written with a digraph though?

https://sl.wikipedia.org/wiki/Ljubljana

Is there a word we can use that does?

(Plus we effectively have an L and two Js in "Ǉjubljana".)

adorilson replied:
My initial suggestion was ǋemačka[0], the word used in the bug report to fix capitalize[1].

[0] https://github.com/python/cpython/pull/105670/files#diff-08a31a70dd1f6d97aa8dacdce77db4de04c700d9949be1af611a595186aad5b3R1628

[1] https://bugs.python.org/msg339568

hugovk replied:
Right, so:
```suggestion
      >>> 'ǋemačka starts with a non-English digraph'.capitalize()
      'ǋemačka starts with a non-english digraph'
```

But I'm not sure if the digraph example adds much here, so shall we skip it?

```suggestion
```

adorilson replied:
> But I'm not sure if the digraph example adds much here, so shall we skip it?

I thought it could be useful when I saw the Changed in version 3.8 note. In general, I also think useful to have examples with non-English or non-ASCII characters.

adorilson replied:
Something was wrong here. I intented to commit the suggestion with 'ǋemačka', but it disappeared. It was applied the suggestion from 23th March.

adorilson committed:
gh-106318: Add an example for str.capitalize()

Co-authored-by: Hugo van Kemenade <1324225+hugovk@users.noreply.github.com>

hugovk commented on Doc/library/stdtypes.rst:
```diff
@@ -1759,7 +1759,14 @@ expression support in the :mod:`re` module).
+      >>> 'PYTHON IS AMAZING'.capitalize()
+      'Python is amazing'
+      >>> 'Ǉjubljana starts With a non-English digraph'.capitalize()
+      'ǈjubljana starts with a non-english digraph'
```
Probably just a mis-click, let's try again (and with "non-ASCII"):

```suggestion
      >>> 'ǋemačka starts With a non-ASCII digraph'.capitalize()
      'ǋemačka starts with a non-ascii digraph'
```

adorilson replied:
It worked. Thank you.

hugovk replied:
Hmm, I noticed my phone (Android with Chrome) renders "ǋ" as a box:

<img width="1080" height="576" alt="image" src="https://github.com/user-attachments/assets/5ceaaa85-b710-4fd8-9b4d-99886a54260c" />

I wonder if there's a more universally supported character?

adorilson replied:
I've tried every digraph listed here: https://en.wikipedia.org/wiki/Digraph_(orthography)#In_Unicode

And all of them have failed. I think doesn't a more universally supported character. But, the problem is only on the interactive shell, on the regular text it is ok.

So, we need to update the font-family. But it is a docs issue or a Sphinx issue?

As I could see, we have:
Interactive shell font-family: Menlo, Consolas, Monaco, Liberation Mono, Lucida Console, monospace;

Regular text font-family: -apple-system, BlinkMacSystemFont, avenir next, avenir, segoe ui, helvetica neue, helvetica, Cantarell, Ubuntu, roboto, noto, arial, sans-serif;

![WhatsApp Image 2025-10-01 at 17 08 25](https://github.com/user-attachments/assets/0cf9c0f1-bfe8-4220-b75a-a020742d98fb)

adorilson replied:
I've reverted the experimental commits.

adorilson committed:
gh-106318: Add an example for str.capitalize()

Co-authored-by: Hugo van Kemenade <1324225+hugovk@users.noreply.github.com>

adorilson committed:
Trying digraphs on stdtypes.rst (should be revert)

See https://github.com/python/cpython/pull/134272#discussion_r2394501827

StanFromIreland commented on Doc/library/stdtypes.rst:
```diff
@@ -1617,7 +1617,7 @@ category.
 |                          |  :meth:`str.casefold`                     |                                                   |
 |                          +-------------------------------------------+---------------------------------------------------+
-|                          |  :meth:`str.capitalize`                   |  :meth:`bytes.capitalize`                         |
+|                          |  :meth:`str.`                   |  :meth:`bytes.capitalize`                         |
```
This seems accidental.

adorilson replied:
Sorry, it was. Anyway, this commit should be reverted.

adorilson committed:
New attemptive with digraphs on stdtypes.rst

adorilson committed:
Recovering capitalize on stdtypes.rst

adorilson committed:
Update stdtypes.rst

adorilson committed:
Revert "Update stdtypes.rst"

This reverts commit fc207585c13b1478e96b6bf24209f3790201359e.

adorilson committed:
Revert "Recovering capitalize on stdtypes.rst"

This reverts commit 6b3052cd28169ba0f142e8fb0c58a7ba8ccf69b6.

adorilson committed:
Revert "New attemptive with digraphs on stdtypes.rst"

This reverts commit 2fd3ca1a7bb04808968ed4e0e3ba5ec5f6cde6ca.

adorilson committed:
Revert "Trying digraphs on stdtypes.rst (should be revert)"

This reverts commit e1f29ec9c3d40922afa98cb30e6f9d7d63136ecb.

adorilson committed:
gh-106318: Add doctest role for str.capitalize() examples

adorilson committed:
Merge branch 'encukou:main' into 106318
